types/google-spreadsheet/index.d.ts

```diff
@@ -1112,7 +1115,7 @@ export class GoogleSpreadsheet implements SpreadsheetBasicProperties {
      *
      * // doc is ready to be used
      */
-    useServiceAccountAuth(credentials: ServiceAccountCredentials): Promise<void>;
+    useServiceAccountAuth(credentials: ServiceAccountCredentials, impersonateAs?: string): Promise<void>;
```

Comment:
This argument defaults to null if it's not passed, so null should probably be accepted as well:

```suggestion
    useServiceAccountAuth(credentials: ServiceAccountCredentials, impersonateAs?: string | null): Promise<void>;
```

Reply:
good catch man. @fujikky with this addition it looks good to go

Reply:
@MysteryBlokHed Thanks for your review, and I've committed to your suggestion!